File: Advanced Methods of Programming/Lab1C-MPP/Lab1C-MPP/Controller/ServiceInscriere.cs
using Lab1C_MPP.Domain;
using Lab1C_MPP.Repo;
using System.Collections.Generic;

namespace Lab1C_MPP.Controller
{
    class ServiceInscriere
    {
        private RepoInscriere repoInscriere;

        public ServiceInscriere(RepoInscriere repoInscriere)
        {
            this.repoInscriere = repoInscriere;
        }

        public void Save(Inscriere inscriere)
        {
            Inscriere i = this.repoInscriere.FindOne(inscriere.GetId());
            if (i == null)
                this.repoInscriere.Save(inscriere);
            else throw new RepositoryException("Participantul este deja inscris la acesta proba!");
        }

        public IEnumerable<Inscriere> findAll()
        {
            return this.repoInscriere.FindAll();
        }

        public Inscriere FindOne(int id) {
            return this.repoInscriere.FindOne(id);
        }

        public int size()
        {
            return this.repoInscriere.Size();
        }
    }
}

File: Advanced Methods of Programming/Tema3JAVA/src/main/java/service/ServiceInscriere.java
package service;

import domain.Inscriere;
import repo.RepoException;
import repo.RepoInscriere;

public class ServiceInscriere {
    private RepoInscriere repoInscriere;

    public ServiceInscriere(RepoInscriere repoInscriere){ this.repoInscriere=repoInscriere; }

    public void save(Inscriere inscriere){
        Inscriere i=this.repoInscriere.findOne(inscriere.getId());
        if(i==null) this.repoInscriere.save(inscriere);
        else throw new RepoException("Participantul este deja inscris la acesta proba!");
    }

    public Iterable<Inscriere> findAll() { return this.repoInscriere.findAll(); }

    public Inscriere findOne(Integer id){ return this.repoInscriere.findOne(id); }

    public Integer size(){ return this.repoInscriere.size(); }
}
